collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java

```diff
@@ -180,13 +199,26 @@ void configureButtons(FormController formController) {
     }
 
     protected void goUpLevel() {
-        Collect.getInstance().getFormController().stepToOuterScreenEvent();
+        FormController formController = Collect.getInstance().getFormController();
+        boolean shouldShowRepeatGroupPicker = repeatGroupPickerIndex != null;
```

Comment:
This is repeated below. Make it a method? Or even just use `repeatGroupPickerIndex != null` inline? The latter seems pretty readable to me...